packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -86,18 +86,31 @@ export class MetadataOutlineProvider
   public async getComponents(
     metadataType: BrowserNode
   ): Promise<BrowserNode[]> {
-    const username = this.defaultOrg!;
-    const cmpUtil = new ComponentUtils();
-    const componentsList = await cmpUtil.loadComponents(
-      username,
-      metadataType.label!
-    );
-    const nodeList = componentsList.map(
-      cmp => new BrowserNode(cmp, NodeType.MetadataCmp)
-    );
-    if (nodeList.length === 0) {
-      nodeList.push(
-        new BrowserNode(nls.localize('empty_components'), NodeType.EmptyNode)
+    let nodeList: BrowserNode[] = [];
+    if (TypeUtils.FOLDER_TYPES.has(metadataType.label!)) {
+      let typeName = metadataType.label!;
+      if (typeName === 'EmailTemplate') {
+        typeName = 'Email';
+      }
+
+      const folders = await this.loadAndMapComponents(
+        `${typeName}Folder`,
+        NodeType.Folder
+      );
+      for (const folder of folders) {
```

Comment:
What's the limit of folders per Metadata Type ? Doing this loop represents 1 query per iteration/folder because we're calling `this.loadAndMapComponents`.